components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1427,9 +1427,22 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         }
         await this.teamDB.addMemberToTeam(user.id, invite.teamId);
         const team = await this.teamDB.findTeamById(invite.teamId);
+
+        await this.ensureTeamsEnabled();
+
         return team!;
     }
 
+    protected async ensureTeamsEnabled() {
+        if (this.user && !this.user?.rolesOrPermissions?.includes('teams-and-projects')) {
+            this.user.rolesOrPermissions = [...(this.user.rolesOrPermissions || []), 'teams-and-projects'];
+            await this.userDB.updateUserPartial({
+                id: this.user.id,
+                rolesOrPermissions: this.user.rolesOrPermissions
+            })
+        }
+    }
```

Comment:
Optional: The common pattern seems to be getting a user via `this.check{,AndBlock}User()` and then using it as a variable. You could potentially pass `user` as an argument to `ensureTeamsEnabled`, and avoid the direct `this.user` access.

Reply:
the purpose of `ensureTeamsEnabled` is to modify the state of `this.user`, therefore this is somewhat exceptional.

Reply:
Makes sense, thanks. Also it should be removed again soon, so resolving this. 😊